graph-service/enricher.py

```
@@ -0,0 +1,875 @@
+            
+            if origin_result and dest_result:
+                origin_id = origin_result["id"]
+                dest_id = dest_result["id"]
+                safe_rel_type = "".join(c if c.isalnum() or c == "_" else "_" for c in rel.type.strip())
+                
+                # Check if relationship already exists
+                check_query = f"""
+                MATCH (a {{id: $origin_id}})-[r]->(b {{id: $dest_id}})
+                WHERE type(r) = $rel_type
+                RETURN count(r) AS exists
+                """
```

Ruff (0.14.7)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)